frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3803,14 +3803,17 @@ const IR::Node* TypeInference::postorder(IR::SwitchStatement* stat) {
         for (auto c : stat->cases) {
             if (c->label->is<IR::DefaultExpression>())
                 continue;
-            auto pe = c->label->to<IR::PathExpression>();
-            CHECK_NULL(pe);
```

Comment:
Wouldn't it be easier to just do:
```
if (pe == nullptr) {
                typeError("%1%: 'switch' label must be an action name or 'default'",
                          c->label);
  continue;
}
```
Also I think the check for default needs something extra because you can have duplicated default still.

Reply:
I have improved the error messages too